experiment/nursery/main.go

```diff
@@ -142,3 +165,48 @@ func cacheHandler(cache *diskcache.Cache) http.Handler {
 		}
 	})
 }
+
+/*
+	--remount related code below
+*/
+
+// exec command and returns lines of combined output
+func commandLines(cmd []string) (lines []string, err error) {
+	c := exec.Command(cmd[0], cmd[1:]...)
+	b, err := c.CombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(string(b), "\n"), nil
+}
+
+// uses `mount -l` to find the device / mountpoint on which path is mounted
+func findMountForPath(path string) (device, mountPoint string, err error) {
+	mounts, err := commandLines([]string{"mount", "-l"})
+	if err != nil {
+		return "", "", err
+	}
+	// these lines are like:
+	// $fs on $mountpoint type $fs_type ($mountopts)
+	device, mountPoint = "", ""
+	for _, mount := range mounts {
+		parts := strings.Fields(mount)
+		if len(parts) >= 3 {
+			currDevice := parts[0]
+			currMount := parts[2]
+			// we want the longest matching mountpoint
+			if strings.HasPrefix(path, currMount) && len(currMount) > len(mountPoint) {
+				device, mountPoint = currDevice, currMount
+			}
+		}
+	}
+	return device, mountPoint, nil
+}
+
+// remounts with 'strictatime,lazyatime'
+func remountWithLazyAtime(device, mountPoint string) error {
+	_, err := commandLines([]string{
+		"mount", "-o", "remount,strictatime,lazytime", device, mountPoint,
```

Comment:
couldn't you use `syscall.Mount` and pass `syscall.MS_REMOUNT | syscall.MS_STRICTATIME | syscall.MS_LAZYTIME` as options?

Reply:
well, I guess go doesn't know about lazytime yet. https://github.com/lxc/lxd/commit/7446f6ba9d34552f592a96e6158699677488f41b

Reply:
though `golang.org/x/sys/unix` has it.

Reply:
I'd also have to parse and plumb through the type, `mount` is a little more magical

Reply:
Discussed this offline, shelling out smells a bit gross but [syscall.Mount also appears to be a wrapper around mount(2) anyhow](https://stackoverflow.com/a/35484096).